src/main/java/homework_4/custom_file_reader/CustomFileReader.java

```diff
@@ -0,0 +1,130 @@
+package homework_4.custom_file_reader;
+
+import java.io.*;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.charset.StandardCharsets;
+
+public class CustomFileReader {
```

Comment:
Opt: unnessessary method boxing and inner try-catches... seems not needed

Reply:
Looks like the code is hard to read without a reason

Reply:
Why all methods are public?

Reply:
Tests ((... How can I test them then?

Reply:
You shouldn't test all methods. If your app is well-incapsulated, it has a few public methods (aka API of your class) and many private methods